### ISSUE TITLE
Enable support for eg a 'fabfile2' module alongside legacy 'fabfile'

### ISSUE THREAD

**Issue by bitprophet** [fabric/fabric#1773]
tl;dr users leaning heavy on the CLI use case (aka...nearly everyone) can have side by side `fab` and `fab2` binaries right now (as per http://docs.fabfile.org/en/2.0/upgrading.html#running-both-fabric-versions-simultaneously), _but_ there's a gaping hole here: both will by default attempt to load a nearby Python module named `fabfile`. Which will necessarily be only v1 or v2 compatible at any given point in time. Bah!

I think a feasible fix for this is to have modern Fabric's loader subclass attempt to load `fabfile2` (or...`fab2file`? HEH) first, and then only fall back to `fabfile` if it couldn't find one. Pros/cons:

- Pro: legacy code works 100% fine and doesn't know or care about any in-progress modern code
- Pro: relatively intuitive...
- Con: ...at least until Fabric 3 comes out
- Con: "wasteful" disk IO, but I cannot imagine that mattering for anyone not running `fab` many times quickly on 5400RPM spinning rust storage, in which case, why aren't they using `execute` + `import` in a single Python process instead??

To help mitigate that last con, we could also honor a flag or env var that disables (or enables?) this behavior.

**Comment by benzkji** [comment]
+1 for only adding this for fab2. reading docs involves understanding more and more things, and the `--collection` option is 100% sufficient, and allows for parallel use of fab and fab2? IMHO.

Also, I've read (more than one time ;-) the upgrade guide - just mentioning `--collection` there would be of great help.

**Comment by bitprophet** [comment]
Is it not mentioned as "the new version of `-f some_other_fabfile.py`"? Cuz it should be.

*checks*

Yea, it is - search for `--collection` in http://www.fabfile.org/upgrading.html#cli-arguments-options-and-behavior, it's got a line item :) (I am sad at how opaque the document can be, but there's just so much info, and no clear way on how to make any of it more accessible...!)

**Comment by benzkji** [comment]
True, thanks!

But if you don't know the old `-f`, you're lost when upgrading ;-) And yes, the document is huge, and the info massive.

**Comment by bitprophet** [comment]
Ooh, I see what you mean! You just meant slapping the above workaround in there as a highlighted "p.s. here's how to run near your old existing v1 compatible fabfile". Yes, that makes sense. I'll probably do that once I take care of this generally.

**Comment by benzkji** [comment]
yes! here: http://www.fabfile.org/upgrading.html#running-both-fabric-versions-simultaneously and/or here http://www.fabfile.org/installing.html#installing-as-fabric2 - as in both places, the really obvious and first change needed to get up and running in parallel mode, is the --collection / -f.
